paperqa/llms.py

```diff
@@ -282,6 +284,37 @@ class QdrantVectorStore(VectorStore):
     vector_name: str | None = Field(default=None)
     _point_ids: set[str] | None = None
 
+    def __init__(self, **data):
+        super().__init__(**data)
+        if isinstance(self.client, AsyncQdrantClient):
+            # Register cleanup for async client
+            atexit.register(self._cleanup_async_client)
+
+    def _cleanup_async_client(self):
+        """Cleanup async client connection."""
+        if isinstance(self.client, AsyncQdrantClient):
+            try:
+                loop = asyncio.get_event_loop()
+                if loop.is_running():
+                    loop.create_task(self.client.close())
+                else:
+                    loop.run_until_complete(self.client.close())
+            except Exception:
+                pass
+
+    async def aclose(self):
+        """Explicitly close async client."""
+        if isinstance(self.client, AsyncQdrantClient):
+            await self.client.close()
+
+    def close(self):
+        """Synchronously close the client."""
+        if isinstance(self.client, AsyncQdrantClient):
+            loop = asyncio.get_event_loop()
+            loop.run_until_complete(self.aclose())
+        elif isinstance(self.client, QdrantClient):
+            self.client.close()
```

Comment:
It's a lot of code to support both `async` and sync workflows. I would rather we just have `AsyncQdrantClient` in the code base.

Can you move to just one of the clients

Reply:
Agreed, will remove the sync client